Yoast/wordpress-seo
# Implementation ai buttons next to seo keyphrase assessments

## Context


* For the following assessments, we need to add an AI button next to the assessments:
   * Keyphrase in introduction
   * Keyphrase subheadings
   * Keyphrase density
   * Keyphrase distribution
* This PR implements the AI button for the assessments above

## Summary


This PR can be summarized in the following changelog entry:

* Adds AI assessment fixes buttons next to the assessment feedback of the _keyphrase in introduction_, _keyphrase in subheading_, and _keyphrase density_ assessments.
* [@yoast/analysis-report] Makes it possible to render the AI Assessment Fixes button inside `AnalysisResult.js`.
* [yoastseo] Adds new property `hasAIFixes` in `AssessmentResult` class.
* [yoastseo] Adds a logic to set `hasAIFixes` to `true` for a certain condition for _keyphrase in introduction_, _keyphrase in subheading_, _keyphrase density_, and _keyphrase distribution_ assessment's result.
* [@yoast/components] Adds components for the AI Assessment Fixes button.
* [wordpress-seo-premium] Adds AI assessment fixes button next to the assessment feedback of _keyphrase in introduction_, _keyphrase in subheading_, _keyphrase density_, and _keyphrase distribution_ assessment.
* [shopify-seo] Adds a wider gap between assessment result list items.

## Relevant technical choices:

* We're aware that we're making changes and using components from `@yoast/components` package. This is because even though technically we can use `@yoast/ui-library` package for this purpose, the stylings of the new AI button will not be consistent with the highlighting and the edit button. For the sake of consistency and UI cohesiveness, we are using the same stylings/components that we're using for the highlighting and the edit button. Eventually, in the new Tailwind style, we need to replace all the buttons and tooltips at once.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

#### WordPress
* Follow the testing instructions as described in [this document](https://docs.google.com/document/d/1ZjfaRaJseoG_bV-O8ghmF5jbAA0bAJLLYAUOA38059w/edit).

#### Shopify
* Test in Shopify product
* Create a product and add some content
* Go to SEO analysis tab
* Confirm that now between assessment result list items, there is a `12px` gap
* Confirm that this is also the case for Readability analysis and Inclusive language analysis
* Additionally, make sure that there is no error in the console

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* The changes introduced in this PR may have an impact on how the eye icon/highlighting button is displayed. So when testing, please also pay attention to how the eye icon is displayed. They should not change.
* We also made some changes in `analysis-report` package which is also being used in Shopify. Please smoke test in shopify and confirm that we didn't change the behaviour/display of the eye icon.

## UI changes


## Other environments


## Documentation


## Quality assurance


## Innovation


Fixes https://github.com/Yoast/wordpress-seo-premium/issues/4285
